src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs

```diff
@@ -99,8 +85,11 @@ internal X509CertificateCollection ClientCertificates
 
         internal void InitializeConnection(string host, int port)
         {
-            _tcpClient.Connect(host, port);
-            _networkStream = _tcpClient.GetStream();
+            lock (this)
```

Comment:
This lock avoids the TcpClient.Connect and TcpClient.Dispose to occur simultaneously.
Alternatively, we may want to make this thread-safe in TcpClient. I can look into that.

Reply:
It makes me nervous that Connect is now being called while holding a lock.  That feels like a recipe for deadlock, or at least potentially elongated hangs.

Reply:
Agree. I'll look at moving this to TcpClient. It may be what is causing the Windows hang...